apps/posthog-proxy/src/nginx.template.conf

```diff
@@ -31,25 +31,31 @@ http {
                 return 204;
             }
 
-            # Add CORS headers to all responses
-            add_header 'Access-Control-Allow-Origin' '$cors_origin' always;
-            add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS' always;
-            add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization' always;
-            add_header 'Access-Control-Allow-Credentials' 'true' always;
-
-            # PostHog requires these headers - route to ingestion API
-            proxy_pass https://eu.i.posthog.com;
-            proxy_redirect off;
-            proxy_ssl_session_reuse off;
-            proxy_ssl_server_name on;
             proxy_set_header Host eu.i.posthog.com;
             proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
             proxy_set_header X-Forwarded-Proto $scheme;
             proxy_set_header X-Real-IP $remote_addr;
-            
+
             # Pass through important headers
             proxy_set_header Referer $http_referer;
             proxy_set_header User-Agent $http_user_agent;
+
+            # Route to ingestion API
+            proxy_pass https://eu.i.posthog.com;
+            proxy_redirect off;
+            proxy_ssl_session_reuse off;
+            proxy_ssl_server_name on;
+
+            # Hide upstream CORS headers to prevent duplicates, then add our own
+            proxy_hide_header Access-Control-Allow-Origin;
+            proxy_hide_header Access-Control-Allow-Methods;
+            proxy_hide_header Access-Control-Allow-Headers;
+            proxy_hide_header Access-Control-Allow-Credentials;
+            proxy_hide_header Access-Control-Max-Age;
```

Comment:
This is the only real change, the rest is just reordering things to reflect the actual order of operations (proxy_set_header sets headers on the request that is sent to the ingestion API, proxy_hide_header removes headers from the response that is received from the ingestion API)